test/jdk/java/nio/Buffer/CharBufferAsCharSequenceTest.java

```diff
@@ -0,0 +1,246 @@
+                        "HeapCharBuffer slice/offset 1 index " + (i - 1) + " to end"));
+
+                int end = buf.length - i;
+
+                buffer = CharBuffer.wrap(buf, i, buf.length - (2 * i));
+                args.add(Arguments.of(buffer, buf, i, end, "HeapCharBuffer index " + i + " to " + end));
+                args.add(Arguments.of(buffer.slice(), buf, i, end, "HeapCharBuffer slice " + i + " to " + end));
+
+                args.add(Arguments.of(CharBuffer.wrap(new String(buf, i, buf.length - (2 * i))), buf, i, end,
+                        "StringCharBuffer index " + i + " to " + end));
+                buffer = CharBuffer.wrap(stringBuf);
+                buffer.position(i);
+                buffer.limit(end);
+                args.add(Arguments.of(buffer.slice(), buf, i, end, "StringCharBuffer slice " + i + " to " + end));
+
+                CharBuffer behbbAsCB = ByteBuffer.allocate(buf.length * 2)
+                                                 .order(ByteOrder.BIG_ENDIAN)
+                                                 .asCharBuffer()
+                                                 .put(buf)
+                                                 .position(1)
+                                                 .slice()
+                                                 .position(i - 1)
+                                                 .limit(end - 1);
+                args.add(Arguments.of(behbbAsCB, buf, i, buf.length - i, "BE HeapByteBuffer as CharBuffer index " + i + " to " + end));
+
+                CharBuffer ledbbAsCB = ByteBuffer.allocateDirect(buf.length * 2)
+                                                 .order(ByteOrder.LITTLE_ENDIAN)
+                                                 .asCharBuffer()
+                                                 .put(buf)
+                                                 .position(1)
+                                                 .slice()
+                                                 .position(i - 1)
+                                                 .limit(end - 1);
+                args.add(Arguments.of(ledbbAsCB, buf, i, buf.length - i, "LE DirectByteBuffer as CharBuffer index " + i + " to " + end));
+            }
+        }
+        return args;
+    }
+
+    @ParameterizedTest
+    @MethodSource("charBufferArguments")
+    void testToString(CharSequence actual, char[] expected, int start, int stop, String description) {
+        assertEquals(new String(expected, start, stop - start), actual.toString(), description);
+    }
+
+    @ParameterizedTest
+    @MethodSource("charBufferArguments")
+    void testLength(CharSequence actual, char[] expected, int start, int stop, String description) {
+        assertEquals(stop - start, actual.length(), description);
+    }
+
+    @ParameterizedTest
+    @MethodSource("charBufferArguments")
+    void testGetChars_range(CharSequence actual, char[] expected, int start, int stop, String description) {
```

Comment:
Another small style point is that typically underscores are not used in (method) names in preference to camel case.